In [16]:
from tensorflow.keras.preprocessing.sequence import pad_sequences
from tensorflow.keras.preprocessing.text import Tokenizer
from sklearn.model_selection import train_test_split
from sklearn.decomposition import PCA
#import plotly.express as px
import tensorflow as tf
import pandas as pd

In [17]:
train_df = pd.read_csv('/home/joaquin/Desktop/Datos/tp2/train.csv')
train_df.head()

id keyword location                                               text  \
0   1     NaN      NaN  Our Deeds are the Reason of this #earthquake M...   
1   4     NaN      NaN             Forest fire near La Ronge Sask. Canada   
2   5     NaN      NaN  All residents asked to 'shelter in place' are ...   
3   6     NaN      NaN  13,000 people receive #wildfires evacuation or...   
4   7     NaN      NaN  Just got sent this photo from Ruby #Alaska as ...   

   target  
0       1  
1       1  
2       1  
3       1  
4       1

In [18]:
#Procesamiento de datos
X_train, X_test, y_train, y_test = train_test_split(train_df['text'], train_df['target'], test_size=0.33,
                                                  random_state=42)

In [50]:
#Parametros
epochs = 5
maxlen = 40
n_words = 300
embedding_dim = 16

In [51]:
#Tokenización
tokenizer = Tokenizer(num_words = n_words, oov_token = '<OOV>')

tokenizer.fit_on_texts(X_train)
word_index = tokenizer.word_index
sequences_train = tokenizer.texts_to_sequences(X_train)
sequences_test = tokenizer.texts_to_sequences(X_test)

padded_train = pad_sequences(sequences_train, maxlen = maxlen, truncating = 'post')
padded_test = pad_sequences(sequences_test, maxlen = maxlen)

In [52]:
#modelo
model = tf.keras.Sequential([tf.keras.layers.Embedding(n_words,embedding_dim, input_length = maxlen),
                             tf.keras.layers.Flatten(),
                             tf.keras.layers.Dense(6, activation='relu'),
                             tf.keras.layers.Dense(1, activation='sigmoid')])

model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
model

In [53]:
#entreno
model.fit(padded_train, y_train, epochs=epochs, validation_data = (padded_test, y_test))

Epoch 1/5
160/160 [==============================] - 3s 18ms/step - loss: 0.6631 - accuracy: 0.5686 - val_loss: 0.6365 - val_accuracy: 0.6546
Epoch 2/5
160/160 [==============================] - 2s 15ms/step - loss: 0.6223 - accuracy: 0.6682 - val_loss: 0.6099 - val_accuracy: 0.7075
Epoch 3/5
160/160 [==============================] - 3s 18ms/step - loss: 0.5825 - accuracy: 0.7243 - val_loss: 0.5829 - val_accuracy: 0.7278
Epoch 4/5
160/160 [==============================] - 3s 16ms/step - loss: 0.5330 - accuracy: 0.7686 - val_loss: 0.5413 - val_accuracy: 0.7421
Epoch 5/5
160/160 [==============================] - 3s 16ms/step - loss: 0.4626 - accuracy: 0.7908 - val_loss: 0.5314 - val_accuracy: 0.7437


Tuning: 
1-  epochs = 10
    maxlen = 50
    n_words = 500
    embedding_dim = 16
Acuracy: 0.7441

2-  epochs = 6
    maxlen = 40
    n_words = 500
    embedding_dim = 16
Accuracy : 0.7656b

3-  epochs = 5
    maxlen = 40
    n_words = 300
    embedding_dim = 16
Accuracy: 0.7437
s